docs/Permissions/Onchain-Permissioning.md

```diff
@@ -0,0 +1,191 @@
+description: Onchain Permissioning
+<!--- END of page meta data -->
+
+# Onchain Permissioning 
+
+Onchain permissioning uses smart contracts to store and maintain the node whitelist. Using onchain permissioning
+enables all nodes to read the node whitelist from one location. 
+                                                                       
+!!! note
+    Pantheon supports onchain permissioning for nodes. Onchain permissioning for accounts will be available 
+    in a future Pantheon release. 
+
+Onchain permissioning uses the following smart contracts: 
+
+* Ingress - a simple contract functioning as a gateway to the Rules contract. The Ingress contract is deployed 
+to a static address. 
+
+* Rules - stores the node whitelist and list of admin accounts. 
+
+## Pre-requisites 
+
+For nodes that are going to set up or maintain the nodes whitelist: 
+
+* [NodeJS](https://nodejs.org/en/) v8.9.4 or later 
+
+* [Truffle](https://truffleframework.com/docs/truffle/getting-started/installation)
+
+## Add Ingress Contract to Genesis File
+
+Add the Ingress contract to the genesis file for your network by copying it from [`genesis.json`](https://github.com/PegaSysEng/permissioning-smart-contracts/blob/master/genesis.json) 
+in the [`permissioning-smart-contracts` repository](https://github.com/PegaSysEng/permissioning-smart-contracts): 
+   
```

Comment:
do we want to explain how to get this code?

Reply:
Yep what Nico said